experiments/ClimaEarth/user_io/arg_parsing.jl

```diff
@@ -0,0 +1,248 @@
+"""
+    get_coupler_config()
+
+Reads in the configuration file and job ID from the command line.
+A dictionary is constructed from the input configuration file, and this
+is returned along with the job ID.
+
+# Returns
+- `config_dict`: A dictionary mapping configuration keys to the specified settings
+"""
+function get_coupler_config()
+    # Read in command line arguments
+    parsed_args = parse_commandline(argparse_settings())
+
+    # Extract the configuration file and job ID
+    config_file = parsed_args["config_file"]
+    job_id = parsed_args["job_id"]
+    # Get the job ID from the config file string if not provided
+    job_id = isnothing(job_id) ? string(split(split(config_file, '/')[end], '.')[1]) : job_id
```

Comment:
Just note that that this function would fail if my config_file were `myconfig` (without the extension).